lib/manager/gitlabci/extract.ts

```diff
@@ -79,3 +82,49 @@ export function extractPackageFile(content: string): PackageFile | null {
   }
   return { deps };
 }
+
+export async function extractAllPackageFiles(
+  config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  const filesToExamine = new Set<string>(packageFiles);
+  const results: PackageFile[] = [];
+
+  // extract all includes from the files
+  for (const file of packageFiles) {
+    const content = await readLocalFile(file, 'utf8');
+    const doc = yaml.safeLoad(content, { json: true }) as any;
+    if (doc?.include && is.array(doc.include)) {
```

Comment:
Extract other deps here and push to results if any deps are found.

Also add a seen files list, so we can't get an endless loop because of circular includes.

Reply:
⬆️